plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeMetadata.java

```diff
@@ -1884,7 +1884,7 @@ private long commitInsertOperation(
             throws IOException
     {
         TrinoFileSystem fileSystem = fileSystemFactory.create(session);
-        long currentVersion = getMandatoryCurrentVersion(fileSystem, handle.location(), handle.readVersion());
+        long currentVersion = getMandatoryCurrentVersion(fileSystem, handle.location(), readVersion.get());
```

Comment:
> Avoid unnecessary IO operations in case of commit retries

Could you leave commit message how this change avoids unnecessary IO operations?